rpc/core/mempool.go

```diff
@@ -42,16 +42,16 @@ func (env *Environment) BroadcastTxSync(ctx *rpctypes.Context, tx types.Tx) (*ct
 	}
 
 	resCh := make(chan *abci.CheckTxResponse, 1)
-	reqRes, err := env.Mempool.CheckTx(tx)
+	reqRes, err := env.Mempool.CheckTx(tx, nil)
 	if err != nil {
 		return nil, err
 	}
-	reqRes.SetCallback(func(_ *abci.Response) {
-		select {
-		case <-ctx.Context().Done():
-		case resCh <- reqRes.Response.GetCheckTx():
-		}
-	})
+	go func() {
+		reqRes.Wait()
```

Comment:
Shouldn't you still the select with a `case` with `ctx.Context().Done()` so it doesn't on `Wait()` when the context is cancelled?

Reply:
Yes. Fixed.